### PR TITLE
[Button] Update default / secondary styles

### LINKED ISSUE #3191: [new DL] Secondary button style styling update
Secondary button style is no longer flat, and now has a slightly raised outline. 

- [ ] Change background color to white

- [ ] Add border

- [ ] Add dropshadow

Screen Shot 2020-08-20 at 8 51 35 AM

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes #3191 and addresses https://github.com/Shopify/polaris-ux/issues/424

image

### WHAT is this pull request doing?

Changes the design of our secondary (default) button without affecting the look of other buttons

Compare against the sort button here: https://www.figma.com/file/DOLJoC3CbldApU26mIIDiK/Page-Component?node-id=61%3A2838

## 

### How to 🎩

http://localhost:6006/?path=/story/all-components-button--all-examples

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)
### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by kyledurand** [comment]
[Chroma link](https://5d559397bae39100201eedc1-bsnjcoafii.chromatic.com/iframe.html?id=all-components-button--all-examples&contexts=New%20Design%20Language%3DEnabled%20-%20Light%20Mode&viewMode=story) for 🎩

File: src/utilities/theme/tokens.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export const Tokens = {
   modalShadow:
     '0px 26px 80px var(--p-shadow-from-dim-light), 0px 0px 1px var(--p-shadow-from-dim-light)',
   topBarShadow: '0 2px 2px -1px var(--p-shadow-from-direct-light)',
+  buttonShadow: '0px 1px 0px rgba(0, 0, 0, 0.05)',
```
Comment (alex-page) [inline src/utilities/theme/tokens.ts:14]: I wonder if this should just be using the `border` css variable? Understanding this is probably not what is defined in Figma.

File: src/components/Select/Select.scss [hunk 2]
```diff
@@ -155,9 +155,11 @@ $stacking-order: (
     right: 0;
     bottom: 0;
     left: 0;
-    border: border-width() solid var(--p-border);
+    border: border-width() solid var(--p-border-neutral-subdued);
+    border-bottom-color: var(--p-border-shadow-subdued);
     border-radius: var(--p-border-radius-base);
-    background-color: var(--p-surface);
+    background: var(--p-surface);
```
Comment (alex-page) [inline src/components/Select/Select.scss:161]: Should we keep it as `background-color`?

**Review by alex-page — APPROVED** [review]

File: src/components/Select/Select.scss (hunk 2 above)
Comment (kyledurand) [reply src/components/Select/Select.scss:161]: I was wondering this as well and was going to ask you about it. I'll double check to make sure this wasn't intentional

Comment (alex-page) [reply src/components/Select/Select.scss]: Thanks Kyle ✨

**Pull request merged**